Source/Core/ClippingPlaneCollection.js

```diff
@@ -376,10 +566,45 @@ define([
     /**
      * The maximum number of supported clipping planes.
      *
+     * @see maxClippingPlanes.glsl
      * @type {number}
      * @constant
      */
-    ClippingPlaneCollection.MAX_CLIPPING_PLANES = 6;
+    ClippingPlaneCollection.MAX_CLIPPING_PLANES = 2048;
+
+    /**
+     * The pixel width of a square, power-of-two RGBA UNSIGNED_BYTE texture
+     * with enough pixels to support MAX_CLIPPING_PLANES.
+     *
+     * A plane is a float in [0, 1) packed to RGBA and an Oct32 quantized normal, so 8 bits or 2 pixels in RGBA
+     *
+     * @type {number}
+     * @constant
+     */
+    ClippingPlaneCollection.TEXTURE_WIDTH = CesiumMath.nextPowerOfTwo(Math.ceil(Math.sqrt(ClippingPlaneCollection.MAX_CLIPPING_PLANES * 2)));
```

Comment:
It might still be better to keep it as a POT though. I think I remember something about those being more efficient for GPU memory or something, although I also vaguely remember someone saying something about how that's not necessarily true anymore...

Reply:
We try to use 1D textures for the 3D Tiles batch table, so there is some precedent there. But it doesn't matter much in the end.